api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/rules/impl/PenRule.java

```diff
@@ -0,0 +1,40 @@
+package ca.bc.gov.educ.studentdatacollection.api.rules.impl;
+
+import ca.bc.gov.educ.studentdatacollection.api.constants.SdcSchoolCollectionStudentValidationFieldCode;
+import ca.bc.gov.educ.studentdatacollection.api.constants.SdcSchoolCollectionStudentValidationIssueSeverityCode;
+import ca.bc.gov.educ.studentdatacollection.api.constants.SdcSchoolCollectionStudentValidationIssueTypeCode;
+import ca.bc.gov.educ.studentdatacollection.api.constants.v1.CollectionTypeCodes;
+import ca.bc.gov.educ.studentdatacollection.api.rules.BaseRule;
+import ca.bc.gov.educ.studentdatacollection.api.service.v1.ValidationRulesService;
+import ca.bc.gov.educ.studentdatacollection.api.struct.SdcStudentSagaData;
+import ca.bc.gov.educ.studentdatacollection.api.struct.v1.SdcSchoolCollectionStudentValidationIssue;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class PenRule implements BaseRule {
+
+    private final ValidationRulesService validationRulesService;
+
+    public PenRule(ValidationRulesService validationRulesService) {
+        this.validationRulesService = validationRulesService;
+    }
+
+    @Override
+    public boolean shouldExecute(SdcStudentSagaData sdcStudentSagaData) {
+        return CollectionTypeCodes.findByValue(sdcStudentSagaData.getCollectionTypeCode(), sdcStudentSagaData.getSchool().getSchoolCategoryCode()).isPresent();
+    }
+
+    @Override
+    public List<SdcSchoolCollectionStudentValidationIssue> executeValidation(SdcStudentSagaData sdcStudentSagaData) {
+        final List<SdcSchoolCollectionStudentValidationIssue> errors = new ArrayList<>();
+        Long penCount = validationRulesService.getDuplicatePenCount(sdcStudentSagaData.getSdcSchoolCollectionStudent().getSdcSchoolCollectionID(), sdcStudentSagaData.getSdcSchoolCollectionStudent().getStudentPen());
+        if(StringUtils.isEmpty(sdcStudentSagaData.getSdcSchoolCollectionStudent().getStudentPen()) || penCount > 1) {
+            errors.add(createValidationIssue(SdcSchoolCollectionStudentValidationIssueSeverityCode.ERROR, SdcSchoolCollectionStudentValidationFieldCode.STUDENT_PEN, SdcSchoolCollectionStudentValidationIssueTypeCode.STUDENT_PEN_DUPLICATE));
+        }
+        return errors;
```

Comment:
Question: say we have two students with duplicate PENS. 

Do we need to flag both as having a PEN_Duplicate issue? 

I think the current logic would log the 2nd student only as having a student_pen_duplicate error.

Reply:
The validation rules runs for one student at a time. It will flag for every student with a duplicate

Reply:
oh cools thanks!